barretenberg/cpp/src/barretenberg/vm/avm/recursion/avm_recursive_verifier.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include "barretenberg/sumcheck/sumcheck.hpp"
+#include "barretenberg/vm/avm/recursion/avm_recursive_flavor.hpp"
+
+namespace bb {
+template <typename Flavor> class AvmRecursiveVerifier_ {
+    using FF = typename Flavor::FF;
+    using BF = typename Flavor::BF;
+    using Curve = typename Flavor::Curve;
+    using Commitment = typename Flavor::Commitment;
+    using CommitmentLabels = typename Flavor::CommitmentLabels;
+    using RelationSeparator = typename Flavor::RelationSeparator;
+
+    using VerificationKey = typename Flavor::VerificationKey;
+
+    using NativeVerificationKey = typename Flavor::NativeVerificationKey;
+    using Builder = typename Flavor::CircuitBuilder;
+    using PCS = typename Flavor::PCS;
+    using Transcript = bb::BaseTranscript<bb::stdlib::recursion::honk::StdlibTranscriptParams<Builder>>;
+    using VerifierCommitments = typename Flavor::VerifierCommitments;
+
+  public:
+    explicit AvmRecursiveVerifier_(Builder* builder,
+                                   const std::shared_ptr<NativeVerificationKey>& native_verification_key);
+    explicit AvmRecursiveVerifier_(Builder* builder, const std::shared_ptr<VerificationKey>& vkey);
+
+    std::array<typename Flavor::GroupElement, 2> verify_proof(const HonkProof& proof);
+
+    std::shared_ptr<VerificationKey> key;
+    std::map<std::string, Commitment> commitments;
```

Comment:
why map? does the order matter? otherwise use unordered_map

Reply:
Actually I can remove these ones. Seem to be unused. I think this might be some legacy code inherited by other recursive verifiers.